Sources/DateScrubber.swift

```diff
@@ -140,7 +200,50 @@ public class DateScrubber: UIViewController {
     }
 
     private func setSectionlabelFrame(){
-        self.sectionLabel.frame = CGRectMake(self.view.frame.width - SectionLabel.RightOffsetForSectionLabel - self.sectionLabel.sectionlabelWidth, 0, self.sectionLabel.sectionlabelWidth, viewHeight)
 
+        let rightOffset = self.sectionLabelState == .Active ? SectionLabel.RightOffsetForActiveSectionLabel : SectionLabel.RightOffsetForInactiveSectionLabel
+        self.sectionLabel.frame = CGRectMake(self.view.frame.width - rightOffset - self.sectionLabel.sectionlabelWidth, 0, self.sectionLabel.sectionlabelWidth, viewHeight)
+    }
+
+    private func setScrubberFrame(){
+        let xPos = self.scrubberState == .Active ? self.containingViewFrame.width -  self.scrubberWidth - DateScrubber.RightEdgeInset : self.containingViewFrame.width
+        scrubberImageView.frame = CGRectMake(xPos, 0, self.scrubberWidth, self.viewHeight)
+    }
+
+    private func setSectionLabelActive(){
+
+        self.sectionLabel.show()
+    }
+
+    private func setSectionLabelInactive() {
+        self.performSelector(#selector(hideSectionLabel), withObject: nil, afterDelay: 3)
+    }
+
+    private func updateSectionLabelFrame() {
+
+        UIView.animateWithDuration(0.2, animations: {
+            self.setSectionlabelFrame()
+        })
+    }
+
+    private func updateDateScrubberFrame() {
+
+        UIView.animateWithDuration(0.2, animations: {
```

Comment:
That's so beautiful! 

![alt text](https://media.giphy.com/media/iJC6vqnoxXvPO/giphy.gif)